src/vs/workbench/parts/files/common/explorerModel.ts

```diff
@@ -70,6 +70,9 @@ export class FileStat implements IFileStat {
 	public hasChildren: boolean;
 	public children: FileStat[];
 	public parent: FileStat;
+	public isVirtualDirectory: boolean;
+	public isVirtualDirectoryMember: boolean;
+	public virtualDirectoryName: string;
```

Comment:
Does it allow to nest file from out of workspace? Should it? What about ignored files?

Reply:
As I would expect ignored files (ignored from VSC) to not show up in this experience. However, I could see a benefit if those ignored files were to be nested. Perhaps an additional setting to show them, if they qualify for nesting, on the behest of the user.